### DISCUSSION TITLE
Abfrageintervall

### DISCUSSION THREAD

**Discussion by mdkeil** [micw/tibber-pulse-reader#21] [Q&A]
Hallo,

erst einmal vielen Dank für das Projekt, ebenso die Hw-Umsetzung mehrerer Schreib-/Leseköpfe. 

Ich nutze schon einige Monate recht problemlos die Variante mit einem Hichi-USB-IR Kopf in Verbindung mit einem parallelen TTL-Lesekopf, wo der Tibber-Pulse dran hängt. Bis letzten Freitag hat dies auch über einige Monate funktioniert-- Aktuell ließt der Pulse die Daten für max. eine Stunde stabil aus und scheint sich dann irgendwie aufzuhängen-- ich muss dann den Pulse neu positionieren dann geht es wieder.. habe bereits unterschiedliche Entfernungen/Positionen ausprobiert, ohne Erfolg.. Heute bekomme ich neue Lithium-Akkus, in der Hoffnung, dass es an der Spannungsversorgung liegt.. sollte dies auch nicht funktionieren, werde ich den Webzugang der Bridge freischalten und mir die Daten lokal holen.. 
Nun die entscheidende Frage: Ich benötige die Daten möglichst in der Aktualisierungsrate vom Zähler, die bei knapp 1s liegt-- liegen die Daten in der gleichen Geschwindigkeit auf der Bridge vor und können entsprechend sekündlich abgerufen werden?

**Comment by micw** [comment]
Hi,
leider kann man maximal alle 2 Sekunden abfragen - ansonsten kommt die Bridge aus dem Tritt. Eine Alternative wäre ein Man-In-The-Middle MQTT-Broker zwischen dem Pulse und dem Internet, dann bekommt man alle Daten.

Ich denke aber dass Du mit den Batterien Glück haben könntest. Ich habe mir AA Dummy-Batterien (Amazon B083KBG7W8) geholt, mit denen ich das Gerät via Netzteil bestomen werde, sobald ich dazu komme oder meine Batterien leer werden.

Ich habe das Projekt übrigens in https://github.com/micw/homedatabroker integriert und werde dieses hier demnächst archivieren.

Viele Grüße,
Michael.

**Reply by mdkeil** [reply]
> Eine Alternative wäre ein Man-In-The-Middle MQTT-Broker zwischen dem Pulse und dem Internet, dann bekommt man alle Daten.

Da bin ich gerade bei, bekomme die Verbindung zu meinem Broker ab irgendwie nicht hin, da sich auf der bridge die Felder ca_cert, certificate und private_key nicht löschen lassen.. dürfen wohl nicht mehr leer sein. 

edit: mit param_set <param> \ lassen sich die entsprechenden Felder löschen.. Verbindung steht nun zum lokalen Broker.. nur die Bridge läuft noch nicht.

**Reply by mdkeil** [reply]
Daten laufen bei Tibber ein über die Bridge.. nur blöde Frage: wenn ich nun mit dem MQTT Explorer auf den Broker schaue, sehe ich keine entsprechenden Topics/Daten.. gibt es einen Trick?

edit: ok, der Trick ist die entprechenden Topics einfach zu abonieren.

**Comment by mdkeil** [comment]
So.. nachdem ich nun den lokalen Broker eingerichtet habe und die Daten über diesen auch zu Tibber gehen nun die Ernüchterung das "nur" alle 3 Sekunden raw SML-Daten vom Zähler kommen und versendet werden.. Werde mir wohl mal [dieses](https://github.com/jsphuebner/esp-egycounter/blob/main/battery-control/tibber/README.md) Projekt anschauen.

**Comment by micw** [comment]
Hehe, mit Link auf mein Blog ;-)
Tatsächlich habe ich da auch schonmal drüber nachgedacht. Allerdings geht das nur dauerhaft stabil, wenn die Änderungen, die Tibber am Protokoll ggf. macht, getrackt werden.

Eine weitere Option, die mir einfällt, wäre den Pulse zu zerlegen, per Netzteil zu bestromen und die SML-Daten direkt einzuspeisen (also den eigentlichen Lesekopf abzutrennen und die Signale direkt, ohne IR-Strecke reinwerfen).

Allerdings läuft der Kram bei mir hier stabil auf der Tibber-Seite. Auf der ESP8266 Seite sind ab und an mal Datenpakete nicht lesbar. Scheint aber in erster Linie am Software-Serial des ESP8266 zu liegen, ein Test mit dem ESP32 (der kann Hardware-Serial auf allen I/Os) bringt wesentlich bessere Ergebnisse.

Ich habe gestern die Boards der nächsten Revision meines Lese/Schreib Kopfes bekommen. Dort habe ich lediglich den Pull-Up von 2k auf 1k verringert und konnte auf dem Steckbrett an einem Lesekopf gleichzeitig zwei Schreibköpfe, einen ESP32 sowie einen USB-Serial-Adapter betreiben und habe auf allen Kanälen ordentlich Daten bekommen. Am Wochenende verteste ich mal die neuen Boards.

Ich habe den Lesekopf vom Pulse übrigens abgenommen (der ist ja flexibel mit Kabel verbaut). Damit sind keine Zugkräfte zwischen Pulse und Schreibkopf, so dass der nicht verrutschen kann

![grafik](https://github.com/micw/tibber-pulse-reader/assets/186753/3a903bc4-e4cf-40ec-a878-aa4e233e148d)

**Comment by mdkeil** [comment]
Was ist denn das weiße zwischen den Unterlegscheiben?.. ich bekomme mit meinem pulse fast gar nichts mehr von meinem parallelen TTL-IR Kopf-- keine Ahnung, wo das Problem liegt-- neue Batterien sind bereits auch drinnen.. setze ich den Pulse direkt auf den Zähler läuft alles wie gehabt.. daher will ich nun ja die Lösung versuchen, die raw-sml-daten vom Hichi-Lesekopf direkt zu schicken, inkl. der Metriken.. Was mir noch nicht ganz klar ist, wenn von Tibber Seite OTAs angestoßen werden.. darauf muss dann ja wahrscheinlich mit geänderten Metriken reagiert werden um zu sagen.. "habe ich umgesetzt"

**Reply by micw** [reply]
> Was ist denn das weiße zwischen den Unterlegscheiben?

Doppelseitiges Klebeband. So ähnlich wie Power-Strips, nur auf Rolle. ich habe eine kleinere Scheibe auf den Schreibkopf geklebt (die größere würde PINs der Stecker berühren) und die größere darüber weil die besser auf den Pulse passt.

**Reply by micw** [reply]
Was setzt du für einen TTL-Lesekopf ein? Hast du den Pulse daran oder am Hitachi?

**Reply by mdkeil** [reply]
Der TTL-IR Kopf ist ebenfalls von Hichi und hängt "parallel" am USB-IR-Kopf, TX wird direkt am Schmitt-Trigger abgenommen.. hat wie gesagt auch über Monate funktioniert-- irgendwie scheint das Signal aber nicht mehr "sauber" zu sein.. Was halt nervt, dass jetzt die Zeit beginnt, wo ich in der Nacht günstig die Autos laden möchte und genau nun streikt die Übertragung..

**Reply by micw** [reply]
Kommen die Daten sauber per TTL bei Dir an?

**Reply by mdkeil** [reply]
muss ich prüfen, wenn ich mal Zeit finde..
